src/System.Private.CoreLib/shared/System/Globalization/Normalization.Unix.cs

```diff
@@ -41,23 +41,23 @@ internal static string Normalize(string strInput, NormalizationForm normalizatio
 
             ValidateArguments(strInput, normalizationForm);
 
-            char[] buf = new char[strInput.Length];
+            var buf = stackalloc char[strInput.Length];
```

Comment:
This is not a safe change; strInput could be arbitrarily long and cause this to stack overflow.

Reply:
@stephentoub indeed do I need to insert a condition to allocate the buffer on heap or stack based on the expected length or or is it not worth it?